File: analyticsservice/src/main/java/com/ridelog/analyticsservice/analytics/AnalyticsService.java
package com.ridelog.analyticsservice.analytics;

import com.ridelog.analyticsservice.analytics.bike.BikeAnalyticsResponse;
import com.ridelog.analyticsservice.analytics.user.UserAnalyticsResponse;

public interface AnalyticsService {

    BikeAnalyticsResponse getBikeAnalytics(
            Long bikeId
    );

    UserAnalyticsResponse getUserAnalytics(
            Long userId
    );
}

File: ridelog-frontend/src/services/analyticsService.js
import axiosClient from "../api/axiosClient";

export const getBikeAnalytics = async (bikeId) => {
  const response = await axiosClient.get(`/analytics/bike/${bikeId}`);

  return response.data;
};

export const getUserAnalytics = async (userId) => {
  const response = await axiosClient.get(`/analytics/user/${userId}`);

  return response.data;
};
